src/gcp_scanner/scanner.py

```diff
@@ -232,13 +234,15 @@ def crawl_loop(initial_sa_tuples: List[Tuple[str, Credentials, List[str]]],
 
       # Get DNS managed zones
       if is_set(scan_config, 'managed_zones'):
-        project_result['managed_zones'] = crawl.get_managed_zones(project_id,
-                                                                  credentials)
+        project_result['managed_zones'] = crawl.get_managed_zones(
+          project_id,
+          ClientFactory.get_client('dns').get_service(credentials),
```

Comment:
Eventually I'd like to move each of these resource-specific client creation calls behind a function that takes the resource string (e.g. "dns") and creds, and leverages a dictionary mapping from resource string to service object. Then we can dynamically update the dict as we loop through the scan_config: either set the newly created object or re-use if it's already there for the resource key.

I'm open to doing this now, or revisiting after the we've implemented each of the clients. WDYT?

Reply:
Hi, I'm trying to gain a complete understanding of the motivation behind this approach. Is it because calling `ClientFactory.get_client('dns').get_service(credentials)` multiple times creates redundant objects in memory? If so I can think of an alternative approach also. :bulb: 

For me, it would be more convenient to incorporate the above refactoring when we implement the double-loop approach. At this moment, I'm having difficulty visualizing the overall implementation, and I think it would be easier for me to grasp if I can start by implementing the simpler parts. Once I have a better understanding of those, I believe I will be able to comprehend this approach more easily :dart:

Reply:
Yes, it helps us keep only one client in memory for each resource.

That sounds perfectly reasonable, let's revisit the idea later and introduce it as an isolated PR if we decide to do it.